core/src/main/scala/org/apache/spark/storage/BlockManagerMaster.scala

```diff
@@ -125,6 +125,16 @@ class BlockManagerMaster(
     driverEndpoint.askSync[Seq[BlockManagerId]](GetPeers(blockManagerId))
   }
 
+  /**
+   * Get list of shuffle service locations available for pushing the shuffle blocks
+   * with push based shuffle
```

Comment:
add . and perhaps add more description about what type of locations are returned